kong/plugins/opentelemetry/handler.lua

```diff
@@ -147,7 +131,39 @@ function OpenTelemetryHandler:access(conf)
   -- Set the sampled flag for the outgoing header's span
   injected_parent_span.should_sample = sampled
 
-  propagation_set(conf.header_type, header_type, injected_parent_span, "w3c")
+  extract_ctx.trace_id      = injected_parent_span.trace_id
+  extract_ctx.span_id       = injected_parent_span.span_id
+  extract_ctx.should_sample = injected_parent_span.should_sample
+  extract_ctx.parent_id     = injected_parent_span.parent_id
+
+  -- return the injected ctx (data to be injected with outgoing tracing headers)
+  return extract_ctx
+end
+
+function OpenTelemetryHandler:access(conf)
+  local root_span = ngx.ctx.KONG_SPANS and ngx.ctx.KONG_SPANS[1]
+
+  -- get the global tracer when available, or instantiate a new one
+  local tracer = kong.tracing.name == "noop" and kong.tracing.new("otel")
+                 or kong.tracing
+
+  -- make propagation work with tracing disabled
+  if not root_span then
+    root_span = tracer.start_span("root")
+    root_span:set_attribute("kong.propagation_only", true)
+
+    -- since tracing is disabled, turn off sampling entirely for this trace
+    kong.ctx.plugin.should_sample = false
+  end
+
+  local injected_parent_span = tracing_context.get_unlinked_span("balancer") or root_span
+
+  propagation.propagate(
+    propagation.get_plugin_params(conf),
+    function(extract_ctx)
+      return get_inject_ctx(conf, extract_ctx, injected_parent_span, root_span, tracer)
+    end
```

Comment:
Creating closure for each request is not friendly for the JIT compiler, could you replace it with a local function?

Reply:
done, thanks